### ISSUE TITLE
Understanding the extended Markdown some outliners use

### ISSUE THREAD

**Issue by scripting** [scripting/Scripting-News#223]
I understand that some outliners use an extended form of Markdown to save their outlines. 

I'm interested in knowing exactly what this format is, because -- if it's easy to support as an export and import format in Drummer, then I want to do it. But first we have to understand what it is. 

A good first step would be to do this: 

1. Find an outliner that supports OPML import, whose native file format is Markdown. 

1a. If you can't find such an app, then be prepared to do some typing! 

2. Import our states.opml example into this app. 

3. Save/export it as Markdown. 

4. Post it here as a comment. 

Everyone but silo-builders wins if our products interop.

**Comment by sabre23t** [comment]
Imported states.opml into Logseq.com v0.3.7
It's converted into states.opml.md attached.
[states.opml.md](https://github.com/scripting/Scripting-News/files/7149719/states.opml.md)

**Comment by scripting** [comment]
@sabre23t -- thanks for the quick response, however -- the link doesn't open.

**Comment by scripting** [comment]
OK -- I've reviewed @sabre23t upload.

0. First, thanks! 👍 

1. Noted that the only head-level attribute that is preserved is &lt;title>. There are quite a few others. 

2. I will come up with a better test, where the individual headlines have attributes. Coming in a minute. I will update the post at the top of the thread.

**Comment by scripting** [comment]
Here's the next test, the OPML source behind the [opml.org](opml.org) home page.

http://opml.org/?format=opml

**Comment by sabre23t** [comment]
> 2. Noted that the only head-level attribute that is preserved is <title>. There are quite a few others.

All the other head level attributes can be supported as custom attributes in Logseq as below. However, currently Logseq doesn't store dateCreated, dateModified, lastCursor in the MD file. The expansionState is stored in the MD file as per block (headline) attribute collapsed:: true.
```
title:: states.opml
dateCreated:: Sun, 03 Dec 2000 01:39:30 GMT
ownerEmail:: dave@userland.com</ownerEmail
description:: The original test outline for OPML apps.
ownerTwitterScreenName:: davewiner
ownerName:: Dave Winer
ownerId:: http://twitter.com/davewiner
urlUpdateSocket:: ws://drummer.scripting.com:1232/
dateModified:: Wed, 18 Aug 2021 18:52:15 GMT
expansionState:: 1,4
lastCursor:: 0

**Comment by scripting** [comment]
@sabre23t -- when you say the "can be" supported as custom attributes, what does that mean? 

Maybe another question I should ask is this -- could you upload an actual file you're working on, without any special prep. I'd like to see what it contains. I may not know what the content is in advance, but hopefully it shouldn't be too hard to discern.

**Comment by sabre23t** [comment]
> Here's the next test, the OPML source behind the [opml.org](opml.org) home page.
> http://opml.org/?format=opml

Imported into Logseq.com v0.3.7 and converted to markdown as 
[opml.org home.md](https://github.com/scripting/Scripting-News/files/7149774/opml.org.home.md)

**Comment by scripting** [comment]
oy. the created atts are gone! ouch.

**Comment by sabre23t** [comment]
> @sabre23t -- when you say the "can be" supported as custom attributes, what does that mean?
Logseq supports custom attributes with almost any name (other than it's own reserved attribute names like `title`). The example from your head-level attributes I gave earlier doesn't clash with any Logseq current attributes, other than title.

> Maybe another question I should ask is this -- could you upload an actual file you're working on, without any special prep. I'd like to see what it contains. I may not know what the content is in advance, but hopefully it shouldn't be too hard to discern.
Attached is my lightly redacted today's journal page `Sep 12th, 2021`.
[2021_09_12.md](https://github.com/scripting/Scripting-News/files/7149792/2021_09_12.md)

**Comment by sabre23t** [comment]
> oy. the created atts are gone! ouch.

Yes. Currently Logseq doesn't support block (headline) level created and modified timestamps. It will later, when Logseq use database not just plain text files (MD or ORG). So I understand. Ref [Tienson Discord message on 30 Jul 2021](https://discord.com/channels/725182569297215569/735747000649252894/870512984491982869).

**Comment by scripting** [comment]
@sabre23t -- thanks you've given me lots to think about. 

The big question is what text is generated by software, and what did you type?

I am sure that "collapsed:: true" was software-generated.

We call that "collapse" in our world, but it's not generated automatically, it's something that applies to rendered text. 

Are the timestamps at the beginning of messages machine-generated? I imagine they are. 

How of this interops with Obsidian, or is this Logseq's language only, or somewhat or ??

**Comment by sabre23t** [comment]
> The big question is what text is generated by software, and what did you type?

Almost all the text in that `2021_09_12.md` I typed. The initial `-` with indentations are automatic based on outline block (headline) level. The timestamp `07:00` etc I actually typed in. But I could also use Logseq `/cu` command to insert current time.

> I am sure that "collapsed:: true" was software-generated.

Yes this is software-generated when you collapse an outline block (headline). But I could also type that block attributes directly.

> How of this interops with Obsidian, or is this Logseq's language only, or somewhat or ??

Generic markdown markups interops with Obsidian. Logseq main extension to Markdown is those attributes `title:: xx`, `collapsed:: true` etc. Obsidian doesn't use those attributes.
